session.go

```diff
@@ -252,12 +255,10 @@ func (s *session) sendLoop() {
 
 	for {
 		select {
-		case frame, open := <-s.out:
-			stillOpen := true
-			if frame != nil {
-				stillOpen = s.send(frame)
-			}
-			if !open || !stillOpen {
+		case <-s.closeCh:
+			return
+		case frame := <-s.out:
```

Comment:
The problem with using a separate channel is that in theory, we could select from closeCh and return even though there's buffered data in s.out remaining to be sent. I'll try a different approach for fixing.

Reply:
How about draining the channel until it becomes empty (both here and in `coalesceAdditionalFrames`)? It feels to me that having a separate close channel made the code a bit more readable.

Reply:
Yes, draining should work.